client/packages/invoices/src/InboundShipment/DetailView/modals/Donor/DonorEditModal.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Autocomplete,
+  Box,
+  useDialog,
+  DialogButton,
+  useNotification,
+  useTranslation,
+  InputWithLabelRow,
+  UpdateDonorMethodInput,
+  Typography,
+  LocaleKey,
+  useConfirmationModal,
+} from '@openmsupply-client/common';
+import { useName } from 'packages/system/src';
+import { getDonorUpdateKeys } from 'packages/system/src/utils';
+import { useInbound } from '../../../api';
+
+interface DonorEditModalProps {
+  invoiceId: string;
+  donorId: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+interface DonorOption {
+  label: string;
+  value: string | null;
+  id: string | null;
+}
+
+interface MethodOption {
+  label: string;
+  value: UpdateDonorMethodInput;
+  message: LocaleKey;
+}
+
+export const DonorEditModal = ({
+  invoiceId,
+  donorId,
+  isOpen,
+  onClose,
+}: DonorEditModalProps) => {
+  const { error } = useNotification();
+  const t = useTranslation();
+  const { Modal } = useDialog({ isOpen, onClose });
+  const { mutateAsync } = useInbound.document.update();
+  const defaultMethod = {
+    label: t(getDonorUpdateKeys(UpdateDonorMethodInput.None).label),
+    value: UpdateDonorMethodInput.None,
+    message: getDonorUpdateKeys(UpdateDonorMethodInput.None).message,
+  };
+
+  const [val, setVal] = useState<DonorOption | undefined>();
+  const [method, setMethod] = useState<MethodOption>(defaultMethod);
+
+  const width = '350px';
+
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('messages.update-all-children-lines'),
+  });
```

Comment:
Maybe it would be helpful also to have the specific update method behaviour displayed in the confirmation modal? 

I wasn't too sure (and also wasn't sure if we want this confirmation modal in the first place)